utils/tests/src/main/java/io/apicurio/registry/utils/tests/LazyRegistryRestClient.java

```diff
@@ -0,0 +1,209 @@
+package io.apicurio.registry.utils.tests;
+
+import java.io.InputStream;
+import java.util.List;
+import io.apicurio.registry.client.RegistryRestClient;
+import io.apicurio.registry.client.RegistryRestClientFactory;
+import io.apicurio.registry.rest.beans.ArtifactMetaData;
+import io.apicurio.registry.rest.beans.ArtifactSearchResults;
+import io.apicurio.registry.rest.beans.EditableMetaData;
+import io.apicurio.registry.rest.beans.IfExistsType;
+import io.apicurio.registry.rest.beans.Rule;
+import io.apicurio.registry.rest.beans.SearchOver;
+import io.apicurio.registry.rest.beans.SortOrder;
+import io.apicurio.registry.rest.beans.UpdateState;
+import io.apicurio.registry.rest.beans.VersionMetaData;
+import io.apicurio.registry.rest.beans.VersionSearchResults;
+import io.apicurio.registry.types.ArtifactType;
+import io.apicurio.registry.types.RuleType;
+
+/**
+ * @author famartin
+ */
+public class LazyRegistryRestClient implements RegistryRestClient {
+
+    private RegistryRestClient instance;
+    private String apiUrl;
+
+    public LazyRegistryRestClient(String apiUrl) {
+        this.apiUrl = apiUrl;
+    }
+
+    private synchronized RegistryRestClient getInstance() {
+        if (instance == null) {
+            instance = RegistryRestClientFactory.create(apiUrl);
+        }
+        return instance;
+    }
```

Comment:
Minor point, but I'd probably call this `delegate` rather than `instance`... unless you make this an actual singleton.